### ISSUE TITLE
aws_route53_targets: SpecRestApi is not convertible to RestApi

### ISSUE THREAD

**Issue by leonaAtkins** [aws/aws-cdk#16817]
### What is the problem?

Using CDK 1.126 (which was listed as containing a fix for this) trying to assign a SpecRestApi as a alias target.

### Reproduction Steps

Create a Spec Rest Api
```
spec_rest_api = aws_apigateway.SpecRestApi(self, "test-api")
```
then try to use it in a route53 target.
```
route53.ARecord(
    self, 
    "AliasRecord",
    zone=zone,
    target=route53.RecordTarget.from_alias(
         aws_route53_targets.ApiGateway(spec_rest_api)
)
```

### What did you expect to happen?

It to create a route53 A record.

### What actually happened?

received the following error

```
jsii.errors.JSIIError: Object of type @aws-cdk/aws-apigateway.SpecRestApi is not convertible to @aws-cdk/aws-apigateway.RestApi
```

### CDK CLI Version

1.126.0 (build f004e1a)

### Framework Version

1.126.0 (build f004e1a)

### Node.js Version

16.4.0

### OS

Windows 10

### Language

Python

### Language Version

3.8.5

### Other information

This is the same as https://github.com/aws/aws-cdk/issues/16227 which I thought was fixed in 1.126?

**Comment by peterwoodworth** [comment]
Hey @leonaAtkins,

I'm not able to reproduce this on the latest version. Are you sure your installed packages are all from the latest version?

**Comment by leonaAtkins** [comment]
> Hey @leonaAtkins,
> 
> I'm not able to reproduce this on the latest version. Are you sure your installed packages are all from the latest version?

Yes, I'm using 1.126 and using Python and still failing with the above error.

Workaround for anyone else facing this.
Add domain to the SpecRestApi and then add this to the aws_route53.ARecord
```
    target=aws_route53.RecordTarget.from_alias(aws_route53_targets.ApiGatewayDomain(self.rest_api.domain_name)),
```
